internal/caveats/builder_test.go

```diff
@@ -228,7 +228,7 @@ func TestMustCaveatExprForTestingWithContext(t *testing.T) {
 
 	contextMap := result.GetCaveat().Context.AsMap()
 	require.Equal(t, "value1", contextMap["key1"])
-	require.Equal(t, float64(42), contextMap["key2"])
+	require.InEpsilon(t, 42, contextMap["key2"], 0.00001)
```

Comment:
I don't understand why using a margin of error here, please revert

Reply:
The changes to InEpsilon came from `float-compare: use require.InEpsilon (or InDelta) (testifylint)`

TLDR since go floats are approximations, this maintains stability and consistency in tests when used with a tiny margin of error. I've never personally seen tests fail due to this approximation, but I've seen InEpsilon used over Equals in various codebases. Probably doesn't really matter, you'd get like one random test failure per million runs in reality.

I've reverted all InEpsilon changes and marked them as `//nolint:testifylint`